core/src/main/java/org/mule/runtime/core/api/event/EventContextFactory.java

```diff
@@ -81,6 +85,24 @@ static EventContext create(String id, String serverId, ComponentLocation locatio
     return create(id, serverId, location, correlationId, empty(), exceptionHandler);
   }
 
+  /**
+   * Builds a new execution context with the given parameters.
+   *
+   * @param flow                                the flow that processes events of this context.
+   * @param location                            the location of the component that received the first message for this context.
+   * @param correlationId                       See {@link EventContext#getCorrelationId()}.
+   * @param externalCompletion                  future that completes when source completes enabling termination of
+   *                                            {@link BaseEventContext} to depend on completion of source.
+   * @param sdkDistributedTraceContextMapGetter the {@link SdkDistributedTraceContextMapGetter} used to retrieve the distributed
+   *                                            trace context.
+   */
+  static EventContext create(FlowConstruct flow, ComponentLocation location, String correlationId,
+                             Optional<CompletableFuture<Void>> externalCompletion,
+                             SdkDistributedTraceContextMapGetter sdkDistributedTraceContextMapGetter) {
```

Comment:
creation of the mule event should not be coupled to an SDK concept. Whatever tracing information is neeeded at the event level should be modeled as part of the event. The SDK must do nothing but provide a facade to that.